hnix-store-core/src/System/Nix/Nar.hs

```diff
@@ -0,0 +1,169 @@
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections       #-}
+{-# LANGUAGE TypeApplications    #-}
+
+{-|
+Description : Allowed effects for interacting with Nar files.
+Maintainer  : Shea Levy <shea@shealevy.com>
+|-}
+module System.Nix.Nar where
```

Comment:
Should we have an explicit export list? Or do we want to export `padLen`?

Reply:
I'll make the export list. `padLen` is shared between `put` and `get` but otherwise is an internal implementation detail.